lib/Doctrine/DBAL/DBALException.php

```diff
@@ -234,13 +234,21 @@ public static function typeExists($name)
      */
     public static function unknownColumnType($name)
     {
-        return new self('Unknown column type "'.$name.'" requested. Any Doctrine type that you use has ' .
-            'to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the ' .
-            'known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database ' .
-            'introspection then you might have forgot to register all database types for a Doctrine Type. Use ' .
-            'AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement ' .
-            'Type#getMappedDatabaseTypes(). If the type name is empty you might ' .
-            'have a problem with the cache or forgot some mapping information.'
+        $bestGuess = '';
+
+        foreach (array_keys(Type::getTypesMap()) as $typeName) {
+            if (levenshtein($typeName, $name) < 3) {
```

Comment:
Sure, sorry for the lack of tests, maybe I should have opened a PR once the code is in a bit more stabilised/tested state. Anyways, will update this soon.